docs/docs/guides/03-typescript.md

```diff
@@ -2,4 +2,226 @@
+};
+
+// books.ts
+const booksService: CommonService = {
+  // ...
+};
+```
+
+### c. Utils
+
+**For type utilities used across different packages**, it is mandatory to place them in their related module's definition file (.d.ts).
+
+- Strapi utils **must** be placed in `@strapi/strapi`'s types module
+- Packages' specific utils **must** be placed in their respective types' module
+
+Basically, utility modules contain reusable type definitions that can be imported wherever needed. This approach promotes code reusability and separation of concerns.
+
+Example:
+
+```typescript
+// core/uid.d.ts
+export type Admin = `admin::${string}`;
+
+// admin.ts
+import type { UID } from '@strapi/strapi';
+
+function getService(uid: UID.Admin): object {
+  // ...
+}
+```
+
+Choose the appropriate approach based on the size and complexity of your feature and the level of reusability of the types.
+
+## 2. Modules, Imports and Exports
+
+### a. Modules
+
+In TypeScript, modules can be used to organize and encapsulate related types and code. Modules provide a way to mimic namespaces and avoid polluting the global scope with numerous type declarations. By grouping related types within modules, it is possible to improve code organization and make it easier to navigate and understand the codebase.
+
+This is especially true when working with packages exports or utilities, where the exported components needs to be scoped.
+
+Example:
+
+```typescript
+// service.d.ts
+export type Admin = Record<string, any>;
+
+// uid.d.ts
+export type Admin = `admin::${string}`;
+
+// index.d.ts
+export * as Service from './service';
+export * as UID from './uid';
+
+// src/index.ts
+import type { Service, UID } from './index';
+
+type AdminService = Record<UID.Admin, Service.Admin>;
+```
+
+### b. Imports
+
+When importing **only types** from a module, `import type` must be used.
+
+The `import type` syntax imports only the types and eliminates the runtime impact of importing the actual values. This helps reduce the bundle size and improves performance by avoiding unnecessary code execution.
+
+Example:
+
+```typescript
+// ❌ don't
+
+import { Attribute, Common } from '@strapi/strapi';
+
+// ✅ do
+
+import type { Attribute, Common } from '@strapi/strapi';
+```
+
+When importing **both types and implementations** from a module:
+
+- `import type` must be used for types
+- `import` must be used for the rest
+
+Example:
+
+```typescript
+// ❌ don't
+
+import { User, findUser } from './user';
+
+const user: User = findUser();
+
+// ✅ do
+
+import type { User } from './user';
+import { findUser } from './user';
+
+const user: User = findUser();
+```
+
+### c. Exports
+
+In most cases, it is recommended to use regular exports (export) instead of the default export (export default). Regular exports provide more flexibility and clarity when importing and using modules. They allow importing specific members of a module explicitly, which improves code readability and maintainability.
```

Comment:
```suggestion
In most cases, it is recommended to use regular exports (`export`) instead of the default export (`export default`). Regular exports provide more flexibility and clarity when importing and using modules. They allow importing specific members of a module explicitly, which improves code readability and maintainability.
```

Reply:
3b8540879cf2f47138fe1855f8f141666d181f4a